common/lib/xmodule/xmodule/modulestore/split_mongo/__init__.py

```diff
@@ -52,22 +112,33 @@ def from_storable(self, stored):
         """
         De-serialize from Mongo-storable format to an object.
         """
+        # Contains the Scope.settings and 'children' field values.
+        # 'children' are stored as a list of (block_type, block_id) pairs.
         self.fields = stored.get('fields', {})
+
+        # XBlock type ID.
         self.block_type = stored.get('block_type', None)
+
+        # DB id of the record containing the content of this XBlock.
         self.definition = stored.get('definition', None)
+
+        # Scope.settings default values copied from a template block (used e.g. when
+        # blocks are copied from a library to a course)
         self.defaults = stored.get('defaults', {})
-        self.edit_info = stored.get('edit_info', {})
 
-    def get(self, key, *args, **kwargs):
-        """
-        Dict-like 'get' method. Raises AttributeError if requesting non-existent attribute and no default.
-        """
-        if len(args) > 0:
-            return getattr(self, key, args[0])
-        elif 'default' in kwargs:
-            return getattr(self, key, kwargs['default'])
-        else:
-            return getattr(self, key)
+        # EditInfo object containing all versioning/editing data.
+        self.edit_info = EditInfo(stored.get('edit_info', {}))
+
+    def __str__(self):
+        return ("BlockData(fields={0.fields}, "
+                "block_type={0.block_type}, "
+                "definition={0.definition}, "
+                "definition_loaded={0.definition_loaded}, "
+                "defaults={0.defaults}, "
+                "edit_info={0.edit_info})").format(self)
+
+    def __contains__(self, item):
```

Comment:
The two case (field or XBlock) seem like they could be handled separately. In particular, if we have an XBlock, then we use `xblock.fields`, and if we have a BlockData, then we use `block_data.fields`. In each case, we get a dict that we can use `in` to check against.

In fact, doing it that way makes the code even smaller, because you're calling `.fields` in both branches (rather than only in one like it does now).